File: static/code/sample/tethered/gdl-seprsr-b.py
# In this sample:
# Display pressure in kPa and PSI units

import time
from DUELink.DUELinkController import DUELinkController

availablePort = DUELinkController.GetConnectionPort()
duelink = DUELinkController(availablePort)

# Methods
def ReadPSI():
    return duelink.Engine.ExecuteCommand("PSI()")

def ReadkPa():
    return duelink.Engine.ExecuteCommand("kPa()")

while True:
    print(f"PSI: {ReadPSI()}")
    print(f"kPa: {ReadkPa()}")
    
    time.sleep(1)  # Sleep for 1 second

File: static/code/sample/tethered/gdl-seprsr-b.js
// In this sample:
// Display pressure in kPa and PSI units

import pkg_serialusb from 'dlserialusb';
const {SerialUSB} = pkg_serialusb

import pkg_duelink from 'duelink';
const {DUELinkController} = pkg_duelink

let duelink = new DUELinkController(new SerialUSB());
await duelink.Connect();

// Sleep helper (ms)
function sleep(ms) {
    return new Promise(resolve => setTimeout(resolve, ms));
}

// Methods
async function ReadPSI() {
    return await duelink.Engine.ExecuteCommand("PSI()");
}

async function ReadkPa() {
    return await duelink.Engine.ExecuteCommand("kPa()");
}

(async function main() {
    while (true) {
        console.log(`PSI: ${await ReadPSI()}`);
        console.log(`kPa: ${await ReadkPa()}`);
        
        await sleep(1000); // Sleep for 1 second
    }
})();